### ISSUE TITLE
Remove requirement to specify path with per-HTTP-method methods.

### ISSUE THREAD

**Issue by garretwilson** [mkotsur/restito#64]
I see that `Condition` has convenience methods for specific HTTP methods, such as `Condition.get(String uri)`. That's fine if I want to specify an exact path. But what if I want to use that in conjunction with `Condition.matchesUri(Pattern p)`? Why isn't there a `Condition.get()` method I can use like this?

```java
whenHttp(server).matches(get(), matchesUri(Pattern.compile("a?b+c*")))…
```

Why can't I just use `get()` by itself, to combine it with any matcher I want?

You might say I could just use `Condition.method(Method m)`, but as explained in #22 that forces me to use the Glassfish API --- and there's no need for that. If there is a `Condition.get()` method at all, why is it tied to a URI, when I could simply say `matches(get(), uri("foo"))` to do the same thing as `matches(get("foo"))`?

Yes, I understand that `get(String uri)` is a convenience method, but convenience methods should allow the long, more flexible form as well. Put another way, it is more flexible to allow `matches(get(), uri("foo"))`, and provide `matches(get("foo"))` as an _additional_ convenience. Only providing the latter without the former makes the API less flexible.

**Comment by mkotsur** [comment]
Good point. Same thing - I'd happily accept a PR for that and quickly make a new release. Or I can pick it up based on time availability.

**Comment by garretwilson** [comment]
It's possible that I could have a developer available to address some of these issues (e.g. #65 and the others we've been discussing) in about a month or two; if so, I would be happy to contribute!
